src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -120,7 +120,15 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                 return await RefreshRtOrFailAsync(cancellationToken).ConfigureAwait(false);
             }
             catch (MsalServiceException e)
-            {
+            {           
+                //Remove the account from cache in case of bad_token sub error
+                if (MsalError.BadToken.Equals(e.SubError, StringComparison.OrdinalIgnoreCase))
+                {
+                    await CacheManager.TokenCacheInternal.RemoveAccountAsync(AuthenticationRequestParameters.Account, AuthenticationRequestParameters.RequestContext).ConfigureAwait(false);
+                    logger.Warning("Removed the account from cache because of bad_token error.");
+                    throw;
```

Comment:
@bgavrilMS should this throw or allow the code to continue and throw at line 143?

Reply:
I think it's fine, as an MsalUIRequiredException will be thrown back to the user, which makes sense since the RT can't be used